src/main/java/io/aiven/kafka/connect/s3/S3SinkTask.java

```diff
@@ -204,7 +205,8 @@ private String oldFullKey(final SinkRecord record) {
                     record, VariableTemplatePart.Parameter.of("padding", "true")
                 )
             );
-        return prefix + key + config.getCompressionType().extension();
+        final String formatSuffix = FormatType.AVRO.equals(config.getFormatType()) ? ".avro" : "";
+        return prefix + key + formatSuffix + config.getCompressionType().extension();
```

Comment:
nit: we may add a reminder to keep this aligned with `AivenCommonConfig#getFilename`

Reply:
Done.